tutorials/07-implementing-a-physics-plugin.md

```diff
@@ -0,0 +1,174 @@
+We will add a pre-defined feature `GetEngineInfo` to our plugin's
+`HelloWorldFeatureList` as follow:
+
+```cpp
+struct HelloWorldFeatureList : ignition::physics::FeatureList<
+    ignition::physics::GetEngineInfo
+> { };
+```
+
+The plugin will be able to return its physics engine metadata.
+We will now implement our plugin class named `HelloWorldPlugin`
+using the defined `FeatureList` above. The class is inherited from
+\ref ignition::physics::Implements3d "Implements3d" to declare that our plugin's
+`HelloWorldFeatureList` will be in the 3D coordinate system.
+
+```cpp
+class HelloWorldPlugin
+    : public ignition::physics::Implements3d<HelloWorldFeatureList>
+{
+  using Identity = ignition::physics::Identity;
+
+  public: Identity InitiateEngine(std::size_t /*_engineID*/) override
+  {
+    this->engineName = "HelloWorld";
+
+    return this->GenerateIdentity(0);
+  }
+
+  public: std::size_t GetEngineIndex(const Identity &/*_id*/) const override
+  {
+    return 0;
+  }
+
+  public: const std::string &GetEngineName(const Identity &/*_id*/) const override
+  {
+    return this->engineName;
+  }
+
+  std::string engineName;
+};
+```
+
+For now, because we do not have any real physics engines, we will define a dummy
+physics engines inside member function `InitiateEngine` by simply setting
+```cpp
+this->engineName = "HelloWorld"
+```
+and returning the engine object using \ref ignition::physics::Identity. Then, we
+define the metadata getters `GetEngineIndex` and `GetEngineName` for our plugin
+and that's it.
+
+Finally, we only have to register our plugin in Ignition Physics as a physics
+engine by:
+
+```cpp
+IGN_PHYSICS_ADD_PLUGIN(
+    HelloWorldPlugin,
+    ignition::physics::FeaturePolicy3d,
+    HelloWorldFeatureList)
+```
+Note that:
+- The first argument is the name of the class that wraps the physics engine into
+a plugin.
+- The second argument is the `FeaturePolicy` for this plugin, i.e. `FeaturePolicy3d`
+- The third argument is the `FeatureList`, specifying all the features that this
+plugin provides, i.e. `HelloWorldFeatureList`
+
+### Setup CMakeLists.txt for building (Dome release)
+
+Now create a file named `CMakeLists.txt` with your favorite editor and add these
+lines for finding `ign-plugin` and `ign-physics` dependencies for Dome release:
+
+```cmake
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+set(IGN_PLUGIN_VER 1)
+find_package(ignition-plugin${IGN_PLUGIN_VER} 1.1 REQUIRED COMPONENTS all)
+set(IGN_PHYSICS_VER 3)
+find_package(ignition-physics${IGN_PHYSICS_VER} REQUIRED)
+```
+
+After that, add the executable pointing to our plugin file and add linking
+library so that `cmake` can compile it:
+
+```cmake
+add_library(HelloWorldPlugin SHARED HelloWorldPlugin.cc)
+target_link_libraries(HelloWorldPlugin
+  PRIVATE
+    ignition-physics${IGN_PHYSICS_VER}::ignition-physics${IGN_PHYSICS_VER})
+```
+
+## Build and run
+
+### Compile the plugin
+
+Now you can build the plugin by:
```

Comment:
would be nice to point out in which directory the following commands should be run, and what the folder structure looks like at this point